src/main/java/com/bootcamp/demo/service/ScooterService.java

```diff
@@ -0,0 +1,77 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.Scooter;
+import com.bootcamp.demo.service.exception.ServiceException;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+@Service
+public class ScooterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScooterService.class);
+    private static final String COLLECTION_SCOOTERS_PATH = "bookings/databases/scooters";
+    private final Firestore db;
+
+    private ScooterService(Firestore db) {
+        this.db = db;
+    }
+
+    public Scooter findScooterById(String scooterId) {
+        try {
+            DocumentSnapshot scooter = db.collection(COLLECTION_SCOOTERS_PATH)
+                    .document(scooterId)
+                    .get()
+                    .get();
+
+            if (!scooter.exists()) {
+                LOGGER.info("FIND SCOOTER BY ID - service function: Scooter does not exists! scooterId = {}", scooterId);
+                return null;
```

Comment:
In case this log is not needed, you could simplify as 
```
return db.collection(COLLECTION_SCOOTERS_PATH)
        .document(scooterId)
        .get()
        .get()
        .toObject(Scooter.class)
```
Also as a note that calling that directly would not throw an exception when the document does not exist.

Reply:
FIxed.